velox/type/Type.cpp

```diff
@@ -1530,6 +1530,145 @@ StringView TimeType::valueToString(int64_t value, char* const startPos) const {
   return StringView{startPos, kTimeToVarcharRowSize};
 }
 
+int64_t TimeType::valueToTime(const StringView& timeStr) const {
+  size_t pos = 0;
+  int32_t hour = 0, minute = 0, second = 0, millis = 0;
+  const char* data = timeStr.data();
+
+  // Helper function to parse a number from a string at a given position
+  auto parseNumber =
```

Comment:
I do think you can simplify and make it very easy to parse with something like this : 
``` 
// Validate timeStr is right length etc..
try {
 auto hours = std::stoi(timeStr.substr(0, 2)); 
 VELOX_CHECK_EQ(timeStr[2], ':');
 auto minutes = std::stoi(timeStr.substr(3, 5));
 VELOX_CHECK_EQ(timeStr[5], ':');
 auto seconds = std::stoi(timeStr.substr(6, 7));
 if (timeStr[7] == '.') {
  // get optional milliseconds
  }
  } catch (...) {
   VELOX_USER_FAIL("Couldnt parse time " , timeStr);
 }
 ```